salt/pillar/postgres.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+'''
+Retrieve Pillar data by doing a postgres query
+
+:maturity: new
+:depends: psycopg2
+:platform: all
+
+Complete example
+=====================================
+
+.. code-block:: yaml
+
+    postgres:
+      user: 'salt'
+      pass: 'super_secret_password'
+      db: 'salt_db'
```

Comment:
I'd suggest supporting a [connection url](https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING) instead. This provides much more flexibility: my personal favorite option in a URL is `?application_name=` so that it's easy to see where connections are coming from even if a proxy is used.

If separate parameters are to be used instead, supplying the connection `port` is important.